File: clojure/collatz-conjecture/src/collatz_conjecture.clj
(ns collatz-conjecture)

(defn collatz
  "Returns the number of steps for num to reach 1
  according to the Collatz Conjecture."
  [num]
  (cond
    (= 1 num) 0
    (even? num) (inc (collatz (quot num 2)))
    :else (inc (collatz (inc (* 3 num))))))

File: python/collatz-conjecture/collatz_conjecture.py
"""Function to calculate the number of steps in the Collatz conjecture"""

def steps(number):
    if number < 1:
        raise ValueError('Only positive integers are allowed')
    return (
        0 if number == 1 
        else 1 + steps(number // 2 if number % 2 == 0 else number * 3 + 1)
    )
